New docs website / First pass at fixing "doc" pages for component - Part 2 (backing classes)

didoo opened the pull request:
### :pushpin: Summary

This if the second part of a series of tasks that intend to fix most of the obvious JS errors in the documentation pages (mainly due to the conversion from the old "how to" and "showcase" pages, to the new markdown-like format.

👉 Not all the component pages have been fixed, see the list here: https://hashicorp.atlassian.net/browse/HDS-1171

### :hammer_and_wrench: Detailed description

In this PR I have:
- added a "preprocess" rule rule to replace `@model` with `this`
  - with the new architecture based on `field-guide` the markdown files have backing classes as “components” not “controllers”, so there’s no model
- added or updated all the markdown backing classes (in the “moveover” folder)

_⚠️ Notice: I have not regenerated the docs in output, to avoid noise in the PR. I'll regenerate them before merging, once the PR i approved. If you want to test I suggest to get the branch up and running in your local environment and test the website there._

### :link: External links

Main JIRA ticket: https://hashicorp.atlassian.net/browse/HDS-1171
Jira task: https://hashicorp.atlassian.net/browse/HDS-1173

***

### 👀 How to review

👉 Review commit-by-commit or by files changed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

alex-ju reviewed (approved):
You're probably aware, but the first 2 commits are #742. They shall go away when merging the other PR though

didoo commented:
> You're probably aware, but the first 2 commits are #742. They shall go away when merging the other PR though

No, wasn't aware 😀. But these days I've done shenanigans to avoid conflicts and split the tasks to make them more "digestible" (cherry-picking, branching, rebasing, etc) so... I'm not surprised. In any case I've rebased and now they've disappeared.
